draft-ietf-jsonpath-base.md

```diff
@@ -517,6 +521,18 @@ This is the concatenation of three lists, two of length one containing `0`, `1`,
 As a consequence of this approach, if any of the selectors selects no nodes,
 then the whole JSONPath selects no nodes.
 
+The current item selector `@` when used outside a filter behaves as the document
+root selector unless a child node is given to the evaluator.
+When a child node is used as current item, evaluator must select it as starting
+node instead of the document root when an expression starting with `@` is
+evaluated.
+
+When a child node is used as current item, the current item selector `@` selects
+it as starting node instead of the document root.
```

Comment:
What happens when a child node is used as the current item and the current item selector is being used inside a filter expression? I presume the current (e.g. array) item "wins", but currently the PR doesn't say.

Reply:
The same that happens when doing, $a[?(@.b[?(@.c == "foo")])].
This need to be explained with or without this PR in the filter section.

Reply:
I think it might be helpful to introduce (and define) the idea of scope here.  Otherwise we should restrict against nested filter queries.

Reply:
@gregsdennis: do you think we need to introduce the idea of scope right now, or just wait the filters PR?

Reply:
I would assume that nested filter queries are out for right now.  There's some discussions [somewhere in the comments on another issue](https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-jsonpath/issues/17#issuecomment-698757622) regarding nested filters.  For now this just adds the syntax.  I think it'd better to bring in scope when we start defining filters.  Then we can start looking at the syntax behavior a bit more.

Reply:
Added a reminder [here](https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-jsonpath/issues/17#issuecomment-803584276), I'm resolving this conversation.